src/toil/utils/toilLaunchCluster.py

```diff
@@ -106,6 +108,10 @@ def main() -> None:
                              "the extra security groups do not have the same name as the cluster name.")
     options = parser.parse_args()
     set_logging_from_options(options)
+
+    #TODO implement permission check aws
+    check_policy_warnings(get_allowed_actions())
```

Comment:
I think that `options.zone` needs to feed in here.

Also, if the user passed `--ec2InstanceProfileArn` already, we need different permissions: we don't need to be able to create arbitrary roles and attach policies, but we do need to be able to assign that particular instance profile to instances. So information about that value probably should come through here as well (at least eventually). As a hack we could just skip the check if that option is used.